patches/0082-lavc-qsvenc_av1-add-missed-options.patch

```diff
@@ -0,0 +1,63 @@
+From c4644e1f981dc572f33a9e4a43269f234551bccc Mon Sep 17 00:00:00 2001
+From: Haihao Xiang <haihao.xiang@intel.com>
+Date: Thu, 13 Oct 2022 09:26:16 +0800
+Subject: [PATCH 49/49] lavc/qsvenc_av1: add missed options
+
+QSV AV1 encoder may support -extbrc and -look_ahead_depth
+---
+ doc/encoders.texi       | 6 ++++++
+ libavcodec/qsvenc.c     | 2 ++
+ libavcodec/qsvenc_av1.c | 2 ++
+ 3 files changed, 10 insertions(+)
+
+diff --git a/doc/encoders.texi b/doc/encoders.texi
+index 4848e812f6..c999546e3f 100644
+--- a/doc/encoders.texi
++++ b/doc/encoders.texi
+@@ -3799,6 +3799,12 @@ This flag controls changing of frame type from B to P.
+ 
+ @item @var{b_strategy}
+ This option controls usage of B frames as reference.
++
++@item @var{extbrc}
++Extended bitrate control.
++
++@item @var{look_ahead_depth}
++Depth of look ahead in number frames, available when extbrc option is enabled.
+ @end table
+ 
+ @section snow
+diff --git a/libavcodec/qsvenc.c b/libavcodec/qsvenc.c
+index 09777414f2..6de4811028 100644
+--- a/libavcodec/qsvenc.c
++++ b/libavcodec/qsvenc.c
+@@ -1012,6 +1012,8 @@ static int init_video_param(AVCodecContext *avctx, QSVEncContext *q)
+ 
+             q->extparam_internal[q->nb_extparam_internal++] = (mfxExtBuffer *)&q->extco2;
+         } else if (avctx->codec_id == AV_CODEC_ID_AV1) {
++            if (q->extbrc >= 0)
```

Comment:
I wonder, do we need to start combining AV1 path with AVC and HEVC? would be more visible with the squashed single AV1 patch variant.

Reply:
So far there are lots of options are available for AVC and HEVC only, so we don't combine av1 path with avc and hevc path.

Reply:
ok, was just reminding that this might be considered in a future.

Reply:
Yes, We'll combine them if possible in the future.